### PR TITLE
Remove hometown group

### PR DESCRIPTION
**Pull request by JohnPMurray**
Fixed issue of hometown in group submissions. Passes backend tests and some manual frontend testing. 

Size: S
Hometown is still requested for group submissions. This should be disabled when submit as group is checked.
[https://trello.com/c/RchxZeyu/94-remove-hometown-from-group-submission](https://trello.com/c/RchxZeyu/94-remove-hometown-from-group-submission)

### CODE REVIEW

File: backend/resolvers/mutations/entry.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ const createEntry = (entry, entryType, entryId, t) => {
       return userFindPromise.then(user => {
 
         let userUpdatePromise = Promise.resolve(null)
-        if (user.hometown != entry.hometown){
+        if (!group && user.hometown != entry.hometown){
```
Comment (JohnPMurray) [inline backend/resolvers/mutations/entry.js:45]: only checks for hometown if its an individual submission

File: frontend/src/Student/components/OtherMediaSubmissionForm.js [hunk 2]
```diff
@@ -205,7 +205,10 @@ class OtherSubmissionForm extends Component {
             }),
             title: yup.string().required('Required'),
             comment: yup.string(),
-            hometown: yup.string().required('Required'),
+            hometown: yup.string().when('submittingAsGroup', {
+              is: 'no',
+              then: yup.string().required('Required')
+            }),
```
Comment (JohnPMurray) [inline frontend/src/Student/components/OtherMediaSubmissionForm.js:211]: Only requires hometown if its an individual submission

File: frontend/src/Student/components/OtherMediaSubmissionForm.js [hunk 3]
```diff
@@ -226,7 +229,9 @@ class OtherSubmissionForm extends Component {
                       participants: values.groupParticipants
                     }
                     : null,
-                hometown: values.hometown,
+                hometown: values.submittingAsGroup === 'no'?  
+                  values.hometown
+                  : null,
```
Comment (JohnPMurray) [inline frontend/src/Student/components/OtherMediaSubmissionForm.js:234]: Only submits a value for hometown if its an individual submission

File: frontend/src/Student/components/OtherMediaSubmissionForm.js [hunk 4]
```diff
@@ -324,13 +331,15 @@ class OtherSubmissionForm extends Component {
                     />
                     {this.renderErrors(touched, errors, 'comment')}
                   </FormGroup>
+                  {values.submittingAsGroup === 'no' ? (
```
Comment (JohnPMurray) [inline frontend/src/Student/components/OtherMediaSubmissionForm.js:334]: disable the hometown field if the submit as group box is checked

File: frontend/src/Student/components/OtherMediaSubmissionForm.js [hunk 5]
```diff
@@ -245,7 +250,9 @@ class OtherSubmissionForm extends Component {
             // Create an entry, show the success modal, and then go to the dashboard
             create(input)
               .then(()=>{
-                handleHometown(values.hometown)
+                if (values.submittingAsGroup == 'no'){
+                  handleHometown(values.hometown)
+                }
```
Comment (JohnPMurray) [inline frontend/src/Student/components/OtherMediaSubmissionForm.js:255]: only perform the handlehometown function on individual submissions.

File: frontend/src/Student/components/PhotoSubmissionForm.js [hunk 6]
```diff
@@ -199,7 +199,10 @@ class PhotoSubmissionForm extends Component {
             }),
```
Comment (JohnPMurray) [inline frontend/src/Student/components/PhotoSubmissionForm.js:199]: Should be the same changes for photo and video

File: frontend/src/Student/components/OtherMediaSubmissionForm.js (hunk 3 above)
Comment (johnbyrne727rit) [reply frontend/src/Student/components/OtherMediaSubmissionForm.js:234]: I think Formik is null tolerant enough that this is what happens anyway without the ternary, but I like that this is explicit about it.

**Review by johnbyrne727rit — APPROVED** [review]
Looks good to me!

File: frontend/src/Student/components/OtherMediaSubmissionForm.js (hunk 3 above)
Comment (JohnPMurray) [reply frontend/src/Student/components/OtherMediaSubmissionForm.js:234]: this is how they did the other group submission fields, I figure its good to keep it consistent if nothing else

**Pull request merged**